test/integration/targets/var_precedence/runme.sh

```diff
@@ -2,6 +2,23 @@
 
 set -eux
 
+MYTMPDIR=$(mktemp -d 2>/dev/null || mktemp -d -t 'mytmpdir')
+trap 'rm -rf "${MYTMPDIR}"' EXIT
+
+# no fact caching
 ansible-playbook test_var_precedence.yml -i ../../inventory -v "$@" \
     -e 'extra_var=extra_var' \
-    -e 'extra_var_override=extra_var_override'
+    -e 'extra_var_override=extra_var_override' \
+    -e '{ write_cache: False }'
+
+# use fact cache, and invoke set_fact with 'cacheable=true' so we can test it across plays in a playbook
+ANSIBLE_CACHE_PLUGIN=jsonfile ANSIBLE_CACHE_PLUGIN_CONNECTION="${MYTMPDIR}" ansible-playbook test_var_precedence.yml -i ../../inventory -v "$@" \
```

Comment:
doesn't the previous entry already do the check w/o caching? .. since it does not enable a cache plugin.

Reply:
Yeah, it occurred to me later that at least one of these is redundant. Probably just needs the default with mem only fact cache that does not use 'cacheable' set_fact, and then one that does. Possibly only one playbook invocation if it runs a play that set_fact's, then a play that uses that fact, then 'meta: clear_facts' and another play to verify the cache is not used.